components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -273,6 +274,8 @@ protected void activate(ComponentContext context) {
             bundleContext.registerService(OAuth2Service.class.getName(), new OAuth2Service(), null);
             // Registering OAuth2ScopeService as a OSGIService
             bundleContext.registerService(OAuth2ScopeService.class.getName(), new OAuth2ScopeService(), null);
+            // Registering OAuth2ScopeService under ScopeService interface as the default service.
+            bundleContext.registerService(ScopeMetadataService.class, new OAuth2ScopeService(), null);
```

Comment:
Shall we use the same instance of the OAuth2ScopeService.class here? since its actually the same service no reason to maintain two instances